Suggested correction for padding

olape-git opened the pull request:
Suggested correction for padding
https://github.com/TN03/expandcontract/issues/16

olape-git committed:
Suggested correction for padding

Suggested correction for padding

TN03 commented:
Puh. 
Also, Content-Padding auf vier Einzelwerte begrenzen: okay, kann man, muss man aber nicht. Es geht dadurch nichts "kaputt" (Laufzeit- / JS-Fehler etc.. Aber gut, macht Sinn.
Zum Rest muss ich fragen, was denn da nun besser läuft bzw. was **korrigiert** wird?
Ich frage mich, ob weiteres RegExpen wirklich notwendig / performanter ist. Für mich persönlich macht es den Code nur schwieriger da ich, soweit möglich, Regular Expressions lieber vermeide. Da geht im Zweifel all zu schnell etwas schief, siehe: https://github.com/TN03/expandcontract/issues/32 . Und in der Regel sind die str* - Funktionen auch performanter und  *berechenbarer*.

Also was genau funktioniert mit der vorherigen Version nicht? 

Richtig cool für **mich** wäre, wenn du die Filter und Parameter etwas genauer kommentieren / beschreiben könntest (ja, ich weiß: RTFM). Ich muss mir sonst aber wirklich sehr viel in der Doku zusammen suchen, wobei dann noch offen ist, ob ich alles richtig überblicke (Verschachtelungen etc.).

olape-git commented:
Ok, wie ich in den Kommentaren schon versucht habe zu beschreiben.
Es wird auf max. 4 Werte begrenzt.
Das ginge sicher auch anders. Man könnte i.e. als case5 einen Fehlermeldung ausgeben.

An regex um die Eingabe auf xxx.px zu prüfen hat sich nichts geändert.
Das ist so von dir übernommen, nur in die Hilfsfunktion verschoben.

Es gibt jetzt wieder die Möglichkeit off. Ich finde das, gerade da wo eh schon fast nur Ziffern einzugeben sind, einfach vielsagender als 0.

Und zusätzlich ist, dass falsche Eingaben in der config, in dem Fall Trennung mit mehr als einem Leerzeichen, per regex abgefangen werden.

Letzteres würde durch preg_split entfallen können, wie auch einige Aufrufe von cleanTinySpaces.

Es sieht nur viel aus, weil ich blöderweise die Änderung eines Funktionsnamens mit reinbekommen habe.

TN03 commented:
Ahh soo :). Ich war echt etwas überfordert mit den ganzen Änderungen. 

> cleanTinySpaces

Das würde ich nicht am Tiny festmachen. Macht der CK vielleicht auch so. Ich muss mal sehen, ob ich eine lauffähige Version für's aktuelle XH und PHP8 habe. Der alte Name war da doch "geeigneter" mMn.


> Es wird auf max. 4 Werte begrenzt.

Ja, hab' ich kapiert. Können wir machen - auch wenn ich persönlich es für komplett unnötig halte (siehe Kommentare im anderen Issue).


> Das ist so von dir übernommen, nur in die Hilfsfunktion verschoben.

Okay. Wird aber eigentlich sonst nirgendwo gebraucht, oder? Und wenn ich es richtig sehe, gibt es dafür dann zusätzlich einen neuen Funktionsparameter.


> Und zusätzlich ist, dass falsche Eingaben in der config, in dem Fall Trennung mit mehr als einem Leerzeichen, per regex abgefangen werden.

Ich halte es für unnötig dafür eine RegExp abzufeuern.


> Letzteres würde durch preg_split entfallen können, wie auch einige Aufrufe von cleanTinySpaces.

Das kann ich derzeit noch nicht überblicken. Macht vielleicht aber Sinn, falls es wirklich noch Probleme mit Leerzeichen geben sollte.

Ich schau' mir das morgen alles noch einmal in Ruhe als kompletten Code an. 
Hast du auf dieser Basis schon Tests für die relevanten / geänderten Funktionen gemacht?
Ansonsten schieben wir es vielleicht besser erst einmal in einen separaten Branch zum Testen.

olape-git committed:
On, ON, Off, OFF in pages, headlines do not work

On, ON, Off, OFF in pages or headlines do not work
i.e. {{{expand('pages = Office, Onboard', 'headlines = Offiziell Mist, Ontario'}}}

https://github.com/TN03/expandcontract/issues/32

olape-git committed:
preg_split instead of explode

'10px    20px     30px    40px'  --> '10px 20px 30px 40px' for for parameters and config

https://github.com/TN03/expandcontract/issues/16

olape-git committed:
Merge pull request #5 from olape-git/On,-ON,-Off,-OFF-in-pages-or-headlines-do-not-work

On, ON, Off, OFF in pages, headlines do not work

olape-git committed:
Merge pull request #6 from olape-git/preg_split-instead-of-explode

preg_split instead of explode

TN03 commented:
So. Jetzt konnte ich mir das mal näher anschauen. Sieht alles gut aus und, mit etwas Zeitversatz, auch ganz übersichtlich.
Ich würde das jetzt mal so übernehmen.
Damit wäre ja dann #31 und #32 erledigt :), oder siehst du noch weiteren Bedarf?

Das einzige was mir nicht gefällt ist der Funktionsname mit ""Tiny". Kleinigkeit. Können wir später noch anpassen.

Nochmals Danke für die ganze Arbeit und Erklärungen.

olape-git commented:
> Das einzige was mir nicht gefällt ist der Funktionsname mit ""Tiny". Kleinigkeit. Können wir später noch anpassen.

Jederzeit



> oder siehst du noch weiteren Bedarf?

Erst mal nur eins, vielleicht sollten wir schon mal die ganzen auskommentierten Teile und unnützen Kommentare entfernen.
Ich glaube das macht es dann etwas übersichtlicher.

Ausserdem lese ich gerade ihr habt schon wieder Änderungen am Padding.
Von daher ;-)

TN03 commented:
> Ausserdem lese ich gerade ihr habt schon wieder Änderungen am Padding.
> Von daher ;-)

Leider. Aber vermutlich ist es jetzt auch eine Lösung. Denn wir können uns nicht verrechnen, wenn wir nichts rechnen ;-).

Die Einzelwerte beim Padding brauchen wir dann *eigentlich* auch nicht mehr. Ich würde es aber so lassen wollen, wegen der nun kompletten Validierung.

Und die beiden Issues mach' ich jetzt mal zu. Falls noch was ist, können wir wieder öffnen oder neue aufmachen.

TN03 commented:
Ach ja, wenn das mit dem Padding nun endlich klappt, sind auch alle anderen möglichen Einheiten für "content-padding" möglich.
Das sollte ja aber nun leicht anzupassen sein :)
